server/queues/tasks/ImportMarkdownZipTask.ts

```diff
@@ -169,6 +176,29 @@ export default class ImportMarkdownZipTask extends ImportTask {
             reference
           );
       }
+
+      const basePath = path.dirname(document.path);
+
+      // check internal document links in the text and replace them with placeholders.
+      // When persisting, the placeholders will be replaced with the right urls.
+      const internalLinks = [
+        ...document.text.matchAll(/\[[^\]]+\]\(([^)]+\.md)\)/g),
+      ];
+
+      internalLinks.forEach((match) => {
+        const referredDocPath = match[1];
+        const normalizedDocPath = decodeURI(
+          path.normalize(`${basePath}/${referredDocPath}`)
+        );
+
+        const referredDocId = docPathToIdMap.get(normalizedDocPath);
+        if (referredDocId) {
+          document.text = document.text.replace(
+            referredDocPath,
+            `<<${referredDocId}>>`
```

Comment:
This is handled during persistence [here](https://github.com/outline/outline/blob/main/server/queues/tasks/ImportTask.ts#L450) (gets replaced with `doc/{docId}`).
This gets handled in the UI as the app takes care of redirection based on id / slug.
However, when collections with this kind of docs are exported, the internal doc links are not replaced with the corresponding relative URLs.

I'm working on another PR that essentially removes `doc/{docId}` trick and sets the proper doc slug during persistence itself. For backward compatibility, the exporters will handle the `docId` replacement as well.

Reply:
> I'm working on another PR that essentially removes doc/{docId} trick and sets the proper doc slug

👏